### PR TITLE
Cleanup Kubelet static analysis issues

### PR DESCRIPTION
**Pull request by tallclair**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Cleanup the `pkg/kubelet/...` issues identified by staticcheck in https://github.com/kubernetes/kubernetes/pull/81189 

**Which issue(s) this PR fixes**:
https://github.com/kubernetes/kubernetes/issues/36858

**Special notes for your reviewer**:
Used to validate the options in https://github.com/kubernetes/kubernetes/pull/81189. Feel free to wait for 81189 to merge first before reviewing.
/hold

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig node
/priority important-longterm

/assign @yujuhong

### CODE REVIEW

**Review by mattjmcnaughton — COMMENTED** [review]
Cool stuff @tallclair!

Overall, love this direction and after a brief skim, all the changes look like the increase code quality.

Just to confirm, `staticcheck` doesn't actually address the errors correct? It just identifies them?
Would be interested to know ballpark how long it took you to make the fixes for `pkg/kubelet/...`.

**Comment by tallclair** [comment]
@mattjmcnaughton Right, it doesn't address them, you have to manually clean up. I don't have a good estimate because I was multitasking and going back and forth with tweaking the staticcheck script, but my guess is about 1-2 hours. My workflow was to paste the staticcheck errors into emacs, which lets me quickly open each file at the triggered line, which sped it up a lot. I think most editors support that format?

**Comment by mattjmcnaughton** [comment]
On Fri, Aug 09, 2019 at 10:41:06AM -0700, Tim Allclair (St. Clair) wrote:
> @mattjmcnaughton Right, it doesn't address them, you have to manually clean up. I don't have a good estimate because I was multitasking and going back and forth with tweaking the staticcheck script, but my guess is about 1-2 hours. My workflow was to paste the staticcheck errors into emacs, which lets me quickly open each file at the triggered line, which sped it up a lot. I think most editors support that format?
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/kubernetes/kubernetes/pull/81206#issuecomment-520004251

Cool stuff, thanks!

**Comment by yujuhong** [comment]
This looks great! Let me know if this is ready for review.

**Comment by tallclair** [comment]
/hold cancel

Ready for review!

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

good stuff :)

definitely think it makes sense to merge this as soon as possible, before we start getting duplicate prs for fixes in kubelet.

File: pkg/kubelet/pleg/generic.go
Comment (yujuhong) [inline pkg/kubelet/pleg/generic.go]: Why removing the condition?

File: pkg/kubelet/remote/utils.go
Comment (yujuhong) [inline pkg/kubelet/remote/utils.go]: I know we want the error messages to start with a lower-case letter, but changing the variable name for this is weird....

I'm okay with this since adding exceptions may not be worth it.

Comment (tallclair) [reply pkg/kubelet/remote/utils.go]: Woops, this was just me not paying enough attention to my find-replace. We're actually not using this check, it was more an experiment to see what applying it here would do. Given that, I just reverted this file.

File: pkg/kubelet/pleg/generic.go
Comment (tallclair) [reply pkg/kubelet/pleg/generic.go]: `delete` is a no-op if the element isn't in the map, so the condition is redundant.

**Comment by tallclair** [comment]
Rebased. Picked up some new failures from https://github.com/kubernetes/kubernetes/pull/81434, but I opted to just keep them on the blacklist and follow up with a separate PR.

Hoping we can merge this quickly to avoid treading water with new failures...

For `hack/.staticcheckfailures` approval:
/assign @BenTheElder 
(Should we expand the approvers set for the blacklist files?)

**Comment by mattjmcnaughton** [comment]
On Wed, Aug 21, 2019 at 11:02:56AM -0700, Tim Allclair (St. Clair) wrote:
> Rebased. Picked up some new failures from https://github.com/kubernetes/kubernetes/pull/81434, but I opted to just keep them on the blacklist and follow up with a separate PR.
> 
> Hoping we can merge this quickly to avoid treading water with new failures...
> 
> For `hack/.staticcheckfailures` approval:
> /assign @BenTheElder 
> (Should we expand the approvers set for the blacklist files?)
> 
> -- 
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub:
> https://github.com/kubernetes/kubernetes/pull/81206#issuecomment-523580898

Yeah, big +1 to try to get these types of diffs merged as soon as possible :)

Thanks for leading the charge on this - exciting stuff!

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
